src/IndexRegistry.sol

```diff
@@ -117,121 +126,170 @@ contract IndexRegistry is IndexRegistryStorage {
+     * Otherwise, push a new historical entry.
+     */
+    function _updateIndexHistory(
+        uint8 quorumNumber,
+        uint32 index,
+        OperatorUpdate storage lastUpdate,
+        bytes32 newOperatorId
+    ) internal {
+        if (lastUpdate.fromBlockNumber == uint32(block.number)) {
+            lastUpdate.operatorId = newOperatorId;
+        } else {
+            _indexHistory[quorumNumber][index].push(OperatorUpdate({
+                operatorId: newOperatorId,
+                fromBlockNumber: uint32(block.number)
+            }));
+        }
+    }
+
+    /// @notice Returns the most recent operator count update for a quorum
+    /// @dev Reverts if the quorum does not exist (history length == 0)
+    function _latestQuorumUpdate(uint8 quorumNumber) internal view returns (QuorumUpdate storage) {
+        uint256 historyLength = _operatorCountHistory[quorumNumber].length;
+        return _operatorCountHistory[quorumNumber][historyLength - 1];
+    }
+
+    /// @notice Returns the most recent operator id update for an index
+    /// @dev Reverts if the index has never been used (history length == 0)
+    function _latestIndexUpdate(uint8 quorumNumber, uint32 index) internal view returns (OperatorUpdate storage) {
+        uint256 historyLength = _indexHistory[quorumNumber][index].length;
+        return _indexHistory[quorumNumber][index][historyLength - 1];
     }
 
     /**
      * @notice Returns the total number of operators of the service for the given `quorumNumber` at the given `blockNumber`
-     * @dev Returns zero if the @param blockNumber is from before the @param quorumNumber existed, and returns the current number 
-     * of total operators if the @param blockNumber is in the future.
+     * @dev Reverts if the quorum does not exist, or if the blockNumber is from before the quorum existed
      */
-    function _getTotalOperatorsForQuorumAtBlockNumber(
+    function _operatorCountAtBlockNumber(
         uint8 quorumNumber, 
         uint32 blockNumber
     ) internal view returns (uint32){
-        // store list length in memory
-        uint256 totalOperatorsHistoryLength = _totalOperatorsHistory[quorumNumber].length;
-        // if there are no entries in the total operator history, return 0
-        if (totalOperatorsHistoryLength == 0) {
-            return 0;
-        }
-
-        // if `blockNumber` is from before the `quorumNumber` existed, return `0`
-        if (blockNumber < _totalOperatorsHistory[quorumNumber][0].fromBlockNumber) {
-            return 0;
-        }
+        uint256 historyLength = _operatorCountHistory[quorumNumber].length;
+        require(historyLength != 0, "IndexRegistry._operatorCountAtBlockNumber: quorum does not exist");
+        require(
+            blockNumber >= _operatorCountHistory[quorumNumber][0].fromBlockNumber, 
+            "IndexRegistry._operatorCountAtBlockNumber: quorum did not exist at given block number"
+        );
```

Comment:
is this not redundant with the revert block at the end of the function? if we delete this then we should also delete the `// Shouldn't be able to reach this point` comment though

Reply:
Good point, thanks. I removed this initial check and the final check's comment.

Reply:
see [231bbe3](https://github.com/Layr-Labs/eigenlayer-middleware/pull/43/commits/231bbe33b677545dbb092033f0be21fdfc83d501)